backend/src/test/java/de/bund/digitalservice/ris/norms/application/service/TimeMachineTest.java

```diff
@@ -0,0 +1,227 @@
+
+    when(xmlDocumentExtractor.getFirstModification(any()))
+        .thenReturn(firstModificationNodeInAmendingLaw);
+    when(xmlDocumentExtractor.getReplacementText(any())).thenReturn(replacementPair);
+    when(xmlDocumentExtractor.findTargetLawNodeToBeModified(any(), any())).thenReturn(targetNode);
+    when(targetNode.getTextContent()).thenReturn("old");
+
+    // when
+    timeMachine.apply(amendingLawString, targetLawString);
+
+    // then
+    verify(xmlDocumentExtractor, times(1)).getFirstModification(any());
+    verify(xmlDocumentExtractor, times(1)).getReplacementText(any());
+    verify(xmlDocumentExtractor, times(1)).findTargetLawNodeToBeModified(any(), any());
+  }
+
+  @Test
+  void oldTextIsReplacedByNewText() {
+    // given
+    final XmlDocumentExtractor xmlDocumentExtractor = new XmlDocumentExtractor();
+    final TimeMachine timeMachine = new TimeMachine(xmlDocumentExtractor);
+
+    String amendingLawString =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?><amendingLaw><akn:mod>In <akn:ref"
+            + " href=\"eli/bund/bgbl-1/1964/s593/1964-08-05/1/deu/regelungstext-1/two/9-34.xml\">paragraph 2</akn:ref> "
+            + "<akn:quotedText>old</akn:quotedText> with <akn:quotedText>new</akn:quotedText>.</akn:mod></amendingLaw>";
+    String targetLawString =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?><targetLaw><akn:p eId=\"two\">old text</akn:p></targetLaw>";
+
+    // when
+    String result = timeMachine.apply(amendingLawString, targetLawString);
+
+    // then
+    assertThat(result)
+        .isEqualTo(
+            "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?><targetLaw><akn:p eId=\"two\">new text</akn:p></targetLaw>");
+  }
+
+  @Test
+  void XmlProcessingExceptionIsThrownWhenAmendingLawXmlIsInvalid() {
+    // given
+    final XmlDocumentExtractor xmlDocumentExtractor = mock(XmlDocumentExtractor.class);
+    final TimeMachine timeMachine = new TimeMachine(xmlDocumentExtractor);
+    String amendingLawString = "SomeRandomText";
+    String targetLawString = "<?xml version=\"1.0\" encoding=\"UTF-8\"?><targetLaw></targetLaw>";
+
+    // when
+    Throwable thrown = catchThrowable(() -> timeMachine.apply(amendingLawString, targetLawString));
+
+    // then
+    assertThat(thrown).isInstanceOf(XmlProcessingException.class);
+  }
+
+  @Test
+  void XmlProcessingExceptionIsThrownWhenTargetLawXmlIsInvalid() {
+    // given
+    final XmlDocumentExtractor xmlDocumentExtractor = mock(XmlDocumentExtractor.class);
+    final TimeMachine timeMachine = new TimeMachine(xmlDocumentExtractor);
+    String amendingLawString =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?><amendingLaw></amendingLaw>";
+
+    String targetLawString = "randomString";
+
+    // when
+    Throwable thrown = catchThrowable(() -> timeMachine.apply(amendingLawString, targetLawString));
+
+    // then
+    assertThat(thrown).isInstanceOf(XmlProcessingException.class);
+  }
+
+  @ParameterizedTest
+  @ValueSource(
```

Comment:
Btw. this `@ParameterizedTest` was recommended by Sonar. I see the point, but are not too happy about how it looks.